meetings/2022-05-11.md

```diff
@@ -0,0 +1,102 @@
+* ...: If we allow Data Grant based access to resources. We would need to say that access to given resource would be for example at most read. It would require to agree who is the ultimate source of the authorization.
+* ...: There is also a case if data grants grants you an access over multiple resources.
+* EP: I just have a quick comment that while talking about Access Modes added in access token, we don't want to get an access token for every request. We shouldn't require the client to get a new token for every request. I don't think the access token should be specific for each request.
+* MB: Yes, as far as I understand, the ID token defined in Solid OIDC, you only claim the identity of requesting agent and issuer/client... But the authorization evaluation needs to happen on every request.
+* EP: The request would include access token, so the authz server can correlate what it knows about the user.
+* LD: I think it's quite hard to prevent token re-issuance. I'll get back to my implementation [on GitHub](https://github.com/laurensdeb/interoperability). Right now, you need token reissuance with every request.
+* ...: When you get a 401 with a token previously got, you can add claims to your token and get new tokens with lesser overhead but it's quite complex to avoid reissuance. You get a ticket with the access modes requested. The access tokens pass information between AS and RS and also give to the token a JTI (token id). You could speed up the work of the UMA AS.
+* EP: There is an incoming request and then you use a backchannel.
+* LD: Indeed you could do that, but the main issue you have right now is that the CSS has to evaluate which access modes are granted over a specific resource. Which makes token re-issuance a necessity. Unless you defined an opaque token with a token introspection endpoint.
+* EP: That's my kind of point. I think the token should be issued based on the claims pushed. With requests being made, the RS says which modes are allowed for which resources.
+* LD: You could have a session id in UMA specific to a RS and via introspection optimise how modes are granted. When using UMA, you should ideally not have to trigger issuance for every request. But I'm still unsure about how to implement it.
+* EP: With UMA, different resources could be protected by different AS. I think with UMA, you could in theory in 401 for different resource return differen AS usign `as_uri`
+* LD: We want the AS to perform authorization. And it's not specified anywhere.
+* MB: Both WAC and ACP have a very resource centric approach, so you evaluate access based on which resource you try to access rather than which resource a user has access to.
```

Comment:
```suggestion
* MB: Both WAC and ACP have a very resource-centric approach, so you evaluate access based on which resource you try to access, rather than which resource a user has access to.
```